server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -718,6 +733,40 @@ public boolean isRequireAlias() {
         return requireAlias;
     }
 
+    @Override
+    public Index getConcreteWriteIndex(IndexAbstraction ia, Metadata metadata) {
+        if (opType() != DocWriteRequest.OpType.CREATE) {
+            return ia.getWriteIndex();
+        }
+
+        if (ia.getType() != IndexAbstraction.Type.DATA_STREAM) {
+            return ia.getWriteIndex();
+        }
+
+        DataStream dataStream = metadata.dataStreams().get(ia.getName());
+        if (dataStream.isTimeSeries(metadata::index) == false) {
+            return ia.getWriteIndex();
+        }
+
+        try (XContentParser parser = contentType.xContent().createParser(TS_EXTRACT_CONFIG, source().streamInput())) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.nextToken(), parser);
+            ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(), parser);
+            String timestampAsString = parser.text();
+            // TODO: deal with nanos too here.
```

Comment:
We add something (format string) to `MappingMetadata` that indicates how the `@timestamp` field should be parsed. We would need to fetch this from the latest backing index of a data stream.

Alternatively we can add information of how the `@timestamp` field should be parsed to `DataStream` class. Which feels like a better place, since we know this prior to selecting the right backing index based on the `@timestamp` field here.